docs/references/glossary.md

```diff
@@ -0,0 +1,312 @@
+
+## Persona (use cases)
+
+Alternative term for a [User Role](#user-role).
+
+## Plugin
+
+A module in Backstage that adds a feature. All functionality outside of [the Backstage framework](#backstage-framework), even the core features, are implemented as plugins.
+
+## Policy (permission plugin)
+
+A construct that takes in a Backstage user and a [permission](#permission-permission-plugin) and returns a [policy decision](#policy-decision-permission-plugin).
+
+## Policy Decision (permission plugin)
+
+A specific response to a user's request to perform an action on a list of [resources](#resource-permission-plugin). Can be either `Approve`, `Deny` or [`Conditional`](#conditional-decision-permission-plugin).
+
+## Popup
+
+A separate browser window opened on top of the previous one.
+
+## Procedure (use cases)
+
+A set of actions that accomplish a goal, usually as part of a [use case](#Use-Case). A procedure can be high-level, containing other procedures, or can be as simple as a single [task](#Task).
+
+## Query Translators (search plugin)
+
+An abstraction layer between a search engine and the [Backstage Search](#search) backend. Allows for translation into queries against your search engine.
+
+## Refresh token
+
+A special token that an [OAuth](#oauth) client can use to get a new [access token](#access-token) when the latter expires.
+
+https://oauth.net/2/refresh-tokens/
+
+## Resource (catalog plugin)
+
+An [entity](#entity) that represents a piece of physical or virtual infrastructure, for example a database, required by a component. See [the catalog docs](https://backstage.io/docs/features/software-catalog/system-model) for more information.
+
+## Resource (permission plugin)
+
+Not to be confused with [Software Catalog resources](#resource-catalog-plugin). Permission resources represent the objects that users interact with and that can be permissioned.
+
+## Rule (permission plugin)
+
+Rules are predicate-based controls that tap into a [resource](#resource-permission-plugin)'s data.
+
+## Role
+
+See [User Role](#User-Role).
+
+## Scaffolder
+
+Known as [Software Templates](#software-templates).
+
+## Scope
+
+A string that describes a certain type of access that can be granted to a user using OAuth, usually in conjunction with [access tokens](#access-token).
+
+## Search
+
+A Backstage plugin that provides a framework for searching a Backstage [app](#app), including the [Software Catalog](#Software-Catalog) and [TechDocs](#TechDocs). A core feature of Backstage.
+
+## Search Engine (Backstage search)
+
+Existing search technology that [Backstage Search](#search) can take advantage of through its modular design. Lunr is the default search in Backstage Search.
+
+## Software Catalog
+
+A Backstage plugin that provides a framework to keep track of ownership and metadata for any number and type of software [components](#component). A core feature of Backstage.
+
+## Software Templates
```

Comment:
You might want to make this two separate entries. "Software Templates" (plural) is the name of the plugin/feature, so making it singular would be incorrect. "Software Template" is a thing (a project skeleton), and should be listed as a singular noun.

Reply:
Updated in b408b6e28dbee90afe6a0b76d078dff0f3b39747